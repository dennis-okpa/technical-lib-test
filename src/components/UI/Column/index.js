import React from 'react'
import heartIcon from '../../../assets/like.png'
import { getDuration } from '../../../utils/translations/duration'
import {
  Container,
  Thumbnail,
  Caption,
  Duration,
  Name,
  Likes,
  Heart,
} from './styles'

function Column({ data }) {
  const { thumbnailUrl, likes, firstName, lastName, duration } = data

  const time = getDuration(duration)

  return (
    <Container>
      <Thumbnail src={thumbnailUrl} alt="thumbnail" />

      <Duration>{time}</Duration>

      <Caption>
        <Name>
          {firstName} {lastName}
        </Name>
        <Likes>
          {likes} <Heart src={heartIcon} />
        </Likes>
      </Caption>
    </Container>
  )
}
export default Column
