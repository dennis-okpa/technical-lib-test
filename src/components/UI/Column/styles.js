import styled from 'styled-components'

export const Container = styled.div`
  position: relative;
  display: inline-block;
  width: 200px;
  border-radius: 8px;
  margin-right: 16px;
  margin-bottom: 24px;
`

export const Thumbnail = styled.img`
  width: 200px;
  height: 200px;
  border-radius: 8px;
`

export const Caption = styled.div`
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
  margin-top: 8px;
`

export const Name = styled.span`
  display: inline-block;
`

export const Likes = styled.span`
  display: inline-flex;
  flex-flow: row wrap;
`

export const Heart = styled.img`
  width: 20px;
  height: 20px;
  margin-left: 4px;
`

export const Duration = styled.span`
  position: absolute;
  top: 8px;
  left: 8px;
  border-radius: 5px;
  background: #000;
  padding: 4px;
  opacity: 0.6;
  color: #fff;
  font-weight: bold;
`
