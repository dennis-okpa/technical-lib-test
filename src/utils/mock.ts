export const mockLibraryData = {
  duration: 33.2277043024078,
  email: 'Ron.Hamill@gmail.com',
  firstName: 'Brooklyn',
  imageUrl:
    'https://images.pexels.com/photos/8422301/pexels-photo-8422301.jpeg',
  lastName: 'Flatley',
  likes: 88,
  thumbnailUrl:
    'https://images.pexels.com/photos/6994926/pexels-photo-6994926.jpeg?auto=compress&cs=tinysrgb&dpr=1&fit=crop&h=200&w=280',
}
