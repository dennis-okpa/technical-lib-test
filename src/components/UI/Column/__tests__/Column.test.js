import React from 'react'
import Column from '../index'
import { render } from '@testing-library/react'
import { mockLibraryData } from '../../../../utils/mock'

describe('Column', () => {
  it('should render correctly', () => {
    let container = render(<Column data={mockLibraryData} />)
    expect(container).toMatchSnapshot()
  })
})
