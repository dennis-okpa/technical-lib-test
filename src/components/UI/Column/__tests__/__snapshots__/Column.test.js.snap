// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvtL hURMSy"
      >
        <img
          alt="thumbnail"
          class="sc-gsDKAQ dezHmh"
          src="https://images.pexels.com/photos/6994926/pexels-photo-6994926.jpeg?auto=compress&cs=tinysrgb&dpr=1&fit=crop&h=200&w=280"
        />
        <span
          class="sc-gKclnd RzqHz"
        >
          0:34
        </span>
        <div
          class="sc-dkPtRN ldbGUI"
        >
          <span
            class="sc-hKwDye xhfHd"
          >
            Brooklyn
             
            Flatley
          </span>
          <span
            class="sc-eCImPb bTCXka"
          >
            88
             
            <img
              class="sc-jRQBWg geCTQY"
              src="like.png"
            />
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvtL hURMSy"
    >
      <img
        alt="thumbnail"
        class="sc-gsDKAQ dezHmh"
        src="https://images.pexels.com/photos/6994926/pexels-photo-6994926.jpeg?auto=compress&cs=tinysrgb&dpr=1&fit=crop&h=200&w=280"
      />
      <span
        class="sc-gKclnd RzqHz"
      >
        0:34
      </span>
      <div
        class="sc-dkPtRN ldbGUI"
      >
        <span
          class="sc-hKwDye xhfHd"
        >
          Brooklyn
           
          Flatley
        </span>
        <span
          class="sc-eCImPb bTCXka"
        >
          88
           
          <img
            class="sc-jRQBWg geCTQY"
            src="like.png"
          />
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
